zomes/wiki/code/src/section.rs

```diff
@@ -71,19 +63,53 @@ pub fn page_element_def() -> ValidatingEntryType {
         }
     )
 }
-pub fn update_element(address: Address, element: Section2) -> ZomeApiResult<Address> {
-    let old_element = hdk::utils::get_as_type::<Section>(address.clone())?;
+pub fn update_section(address: Address, mut section: Section) -> ZomeApiResult<Address> {
+    let old_section = hdk::utils::get_as_type::<Section>(address.clone())?;
+    let anchor_address_option = old_section.anchor_address;
+    section.anchor_address = anchor_address_option.clone();
+    let new_address = hdk::api::update_entry(section.entry(), &address)?;
+    if let Some(anchor_address) = anchor_address_option {
+        let page_address = hdk::get_links(
+            &anchor_address,
+            LinkMatch::Exactly("anchor->Page"),
+            LinkMatch::Any,
+        )?
+        .addresses()[0]
+            .clone();
+        let page: Page = hdk::utils::get_as_type(page_address.clone())?;
+        let sections = page
+            .clone()
+            .sections
+            .into_iter()
+            .filter_map(|d_address| {
+                if d_address != address {
+                    Some(d_address)
+                } else {
+                    Some(new_address.clone())
+                }
+            })
+            .collect();
 
-    hdk::api::update_entry(
-        Section::from(element, old_element.page_address).entry(),
-        &address,
-    )
+        hdk::api::update_entry(
+            Page::from(page.title.clone(), sections, page.timestamp).entry(),
+            &page_address,
+        )?;
+    };
+    Ok(new_address)
 }
-pub fn delete_element(address: Address) -> ZomeApiResult<String> {
-    let page_address = hdk::utils::get_as_type::<Section>(address.clone())?.page_address;
+pub fn delete_section(address: Address) -> ZomeApiResult<String> {
+    let anchor_address = hdk::utils::get_as_type::<Section>(address.clone())?
+        .anchor_address
+        .unwrap();
```

Comment:
Be very careful with these kinds of unwraps! Do we need to make the anchor address an `Option`?